uportal-war/src/main/java/org/apereo/portal/portlets/lookup/PersonLookupHelperImpl.java

```diff
@@ -464,9 +489,14 @@ protected IPersonAttributes getVisiblePerson(
                         IPermission.VIEW_USER_ACTIVITY,
                         person.getName())) {
 
-            // if the user has permission, filter the person attributes according
-            // to the specified permitted attributes
-            final Map<String, List<Object>> visibleAttributes = new HashMap<String, List<Object>>();
+            // If the user has permission, filter the person attributes according
+            // to the specified permitted attributes;  the collection of permitted
+            // attributes can be different based on whether the user is trying to
+            // access information about him/herself.
+            final Set<String> permittedAttributes = person.getName().equals(principal.getKey())
```

Comment:
On skim I'm surprised that `person.getName()` is being compared with `principal.getKey()`. The `name` property of an `IPersonAttributes` matches against the `key` property of an `IAuthorizationPrincipal`?